nautobot/core/tests/test_events.py

```diff
@@ -83,37 +85,108 @@ def test_publish_events_to_broker(self):
         deregister_event_broker(event_broker)
         deregister_event_broker(event_broker_2)
 
+    @load_event_broker_override_settings(
+        NAUTOBOT_EVENT_BROKERS={
+            "SyslogEventBroker": {
+                "CLASS": "nautobot.core.events.SyslogEventBroker",
+                "TOPICS": {
+                    "INCLUDE": ["nautobot.test.*"],
+                    "EXCLUDE": ["*.no-publish*"],
+                },
+            }
+        }
+    )
     def test_publish_events_to_syslog(self):
-        event_broker = SyslogEventBroker()
-        register_event_broker(event_broker)
-
-        try:
-            with self.assertLogs("nautobot.events.nautobot.test.event") as cm:
-                publish_event(topic="nautobot.test.event", payload={"a": 1})
-            self.assertEqual(cm.output, ["INFO:nautobot.events.nautobot.test.event:" + json.dumps({"a": 1}, indent=4)])
-        finally:
-            deregister_event_broker(event_broker)
-
+        with self.assertLogs("nautobot.events.nautobot.test.event") as cm:
+            publish_event(topic="nautobot.test.event", payload={"a": 1})
+            publish_event(topic="nautobot.test.no-publish.event", payload={"a": 1})
+        # This test assets that only `nautobot.test.event` topic was published
```

Comment:
I don't think this is correct - think you'd want to do a separate test with `assertNoLogs("nautobot.events.nautobot.test.no-publish.event")`

Reply:
`cm.output` is a list of all logs collected, Here im asserting only 1 and correct entire exists.

Reply:
Doesn't the parameter to `assertLogs` apply as a filter on the logs collected though? I wouldn't think this would capture *all* logs from Python.

Reply:
Not all logs from python but all logs inside its context

```python
 with self.assertLogs("nautobot.events.nautobot.test.event") as cm:
            publish_event(topic="nautobot.test.event", payload={"a": 1})
            publish_event(topic="nautobot.test.event.another", payload={"b": 1})
```

In this case: `cm.output` would be:
```json
[
     'INFO:nautobot.events.nautobot.test.event:{\n    "a": 1\n}', 
     'INFO:nautobot.events.nautobot.test.event.another:{\n    "b": 1\n}'
]
```

Since both topics are not excluded

Reply:
Right... my point was that `(nautobot.events.)nautobot.test.no-publish.event` isn't included in the `nautobot.events.nautobot.test.event` context, so it wouldn't be captured even if logged?

Reply:
oooh